scico/test/linop/test_linop.py

```diff
@@ -18,77 +18,24 @@
 from scico.typing import PRNGKey
 
 
-def adjoint_AtA_test(A: linop.LinearOperator, key: Optional[PRNGKey] = None, rtol: float = 1e-4):
-    """Check the validity of A.conj().T as the adjoint for a LinearOperator A
-
-    Compares the quantity sum(x.conj() * A.conj().T @ A @ x) against
-    norm(A @ x)**2.  If the adjoint is correct, these quantities should be equal.
+def adjoint_test(A: linop.LinearOperator, key: Optional[PRNGKey] = None, rtol: float = 1e-4):
+    """Check the validity of A.conj().T as the adjoint for a LinearOperator A.
 
     Args:
         A : LinearOperator to test
         key:  PRNGKey for generating `x`.
         rtol:  Relative tolerance
     """
 
-    # Generate a signal in the domain of A
-    x, key = randn(A.input_shape, dtype=A.input_dtype, key=key)
-
-    Ax = A @ x
-
-    AtAx = A.conj().T @ Ax
-    num = snp.sum(x.conj() * AtAx)
-    den = snp.linalg.norm(Ax) ** 2
-    np.testing.assert_allclose(num / den, 1, rtol=rtol)
-
-    AtAx = A.H @ Ax
-    num = snp.sum(x.conj() * AtAx)
-    den = snp.linalg.norm(Ax) ** 2
-    np.testing.assert_allclose(num / den, 1, rtol=rtol)
-
-    AtAx = A.adj(Ax)
-    num = snp.sum(x.conj() * AtAx)
-    den = snp.linalg.norm(Ax) ** 2
-    np.testing.assert_allclose(num / den, 1, rtol=rtol)
-
-
-def adjoint_AAt_test(A: linop.LinearOperator, key: Optional[PRNGKey] = None, rtol: float = 1e-4):
-    """Check the validity of A as the adjoint for a LinearOperator A.conj().T
-
-    Compares the quantity sum(y.conj() * A @ A.conj().T @ y) against
-    norm(A.conj().T @ y)**2.  If the adjoint is correct, these quantities should be equal.
-
-    Args:
-        A : LinearOperator to test
-        key:  PRNGKey for generating `x`.
-        rtol:  Relative tolerance
-    """
-    # Generate a signal in the domain of A^T
-    y, key = randn(A.output_shape, dtype=A.output_dtype, key=key)
-
-    Aty = A.conj().T @ y
-    AAty = A @ Aty
-    num = snp.sum(y.conj() * AAty)
-    den = snp.linalg.norm(Aty) ** 2
-    np.testing.assert_allclose(num / den, 1, rtol=rtol)
-
-    Aty = A.H @ y
-    AAty = A @ Aty
-    num = snp.sum(y.conj() * AAty)
-    den = snp.linalg.norm(Aty) ** 2
-    np.testing.assert_allclose(num / den, 1, rtol=rtol)
-
-    Aty = A.adj(y)
-    AAty = A @ Aty
-    num = snp.sum(y.conj() * AAty)
-    den = snp.linalg.norm(Aty) ** 2
-    np.testing.assert_allclose(num / den, 1, rtol=rtol)
+    assert linop.valid_adjoint(A, A.H, rtol, key)
```

Comment:
I checked that there are other tests for the equivalence of `A.conj().T`, `A.H`, and `A.adj` and there are, so I think this is fine (and lots cleaner)